react/features/base/tracks/middleware.js

```diff
@@ -38,29 +38,30 @@ MiddlewareRegistry.register(store => next => action => {
         break;
 
     case LIB_DID_INIT:
-        store.dispatch(createLocalTracks());
+        store.dispatch(createInitialLocalTracks());
         break;
 
     case SET_AUDIO_MUTED:
         _setMuted(store, action, MEDIA_TYPE.AUDIO);
         break;
 
     case SET_CAMERA_FACING_MODE: {
-        // XXX Destroy the local video track before creating a new one or
-        // react-native-webrtc may be slow or get stuck when opening a (video)
-        // capturer twice.
+        // XXX The camera facing mode of a MediaStreamTrack can be specified
+        // only at initialization time and then it can only be toggled. So in
+        // order to set the camera facing mode, one may destroy the track and
+        // then initialize a new instance with the new camera facing mode. But
+        // that is inefficient on mobile at least so the following relies on the
+        // fact that there are 2 camera facing modes and merely toggles between
+        // them to (hopefully) get the camera in the specified state.
         const localTrack = _getLocalTrack(store, MEDIA_TYPE.VIDEO);
+        let jitsiTrack;
 
-        if (localTrack) {
-            store.dispatch(_disposeAndRemoveTracks([ localTrack.jitsiTrack ]));
+        if (localTrack
+                && (jitsiTrack = localTrack.jitsiTrack)
+                && jitsiTrack.getCameraFacingMode()
```

Comment:
Thank you, @bbaldino! `const { jitsiTrack } = localTrack;` will not work because `localTrack` may be `null`/`undefined`.